### PR TITLE
Add new missing samples and implements all the tests

### PR DESCRIPTION
**Pull request by ThomyLorenzatti**

### CODE REVIEW

File: src/XMLRawParser.ts [hunk 1]
```diff
@@ -14,7 +19,24 @@ export class XMLRawParser {
     });
   }
 
-  async parseString(xmlData: string): Promise<string> {
-    return this.parser.parseStringPromise(xmlData);
+  private convertBooleans(obj: Data = this.data as Data): void {
+    Object.entries(obj).forEach(([key, value]) => {
+      if (typeof value === 'object' && value !== null) {
+        this.convertBooleans(value as Data);
+      } else if (value === 'true') {
```
Comment (GautierBlandin) [inline src/XMLRawParser.ts:26]: Make a function to check if parsable and a function to parse

File: src/XMLRawParser.ts [hunk 2]
```diff
@@ -1,8 +1,13 @@
+/* eslint-disable no-param-reassign */
 import { Parser } from 'xml2js';
 
+type Data = Record<string, unknown>;
```
[the remark this answers is not in the document]
Comment (ThomyLorenzatti) [reply src/XMLRawParser.ts]: Done

File: src/XMLRawParser.ts [hunk 3]
```diff
@@ -1,8 +1,13 @@
...
+
 export class XMLRawParser {
   private parser: Parser;
 
+  private data: Data | undefined;
```
[the remark this answers is not in the document]
Comment (ThomyLorenzatti) [reply src/XMLRawParser.ts]: Done

File: src/XMLRawParser.ts [hunk 4]
```diff
@@ -14,7 +15,30 @@ export class XMLRawParser {
...
+    Object.entries(obj).forEach(([key, value]) => {
+      if (typeof value === 'object' && value !== null) {
+        this.parseBooleanStringsToBooleanRecursive(value as Record<string, unknown>);
+      } else if (this.isParsableBoolean(value)) {
+        obj[key] = this.parseBoolean(value);
+      }
+    });
+  }
+
+  async parseString(xmlData: string): Promise<Record<string, unknown>> {
```
Comment (GautierBlandin) [inline src/XMLRawParser.ts:37]: Invert the order of functions in the class:
- public async parseString comes first
- then the parseBooleanStringsToBooleanRecursive
- then isParsableBoolean and parseBoolean
The reason for that is that it makes the code more readable by making the class understandable from top to bottom. We start with the most important function of the class (the one we make public), and we go down to a smaller and smaller level.

**Review by GautierBlandin — CHANGES_REQUESTED** [review]
One small change to do, good job !

**Review by GautierBlandin — APPROVED** [review]
Good job !

**Pull request merged**